mtp_api/apps/mtp_auth/tests/test_views.py

```diff
@@ -74,30 +79,332 @@ def test_can_access_my_data_including_managing_prisons(self):
+        )
+
+        for user_item in response.data['results']:
+            user = User.objects.get(username=user_item['username'])
+            self.assertEqual(
+                user.applicationusermapping_set.all()[0].application.client_id,
+                client_id
+            )
+            if hasattr(requester, 'prisonusermapping'):
+                matching_prison = False
+                for prison in requester.prisonusermapping.prisons.all():
+                    if prison in list(user.prisonusermapping.prisons.all()):
+                        matching_prison = True
+                        break
+                self.assertTrue(
+                    matching_prison,
+                    msg='User Admin able to retrieve users without matching prisons'
+                )
+
+    def test_list_users_for_bank_user_admin(self):
+        self._check_list_users_succeeds(
+            self.bank_uas[0],
+            'bank-admin'
+        )
+
+    def test_list_users_for_cashbook_user_admin(self):
+        self._check_list_users_succeeds(
+            self.cashbook_uas[0],
+            'cashbook'
+        )
+
+
+class CreateUserTestCase(APITestCase, AuthTestCaseMixin):
+    fixtures = [
+        'initial_groups.json',
+        'test_prisons.json'
+    ]
+
+    def setUp(self):
+        super().setUp()
+        self.cashbook_uas, self.pla_uas, self.bank_uas = make_test_user_admins()
+
+    def get_url(self):
+        return reverse('user-list')
+
+    def test_normal_user_cannot_create_user(self):
+        _, _, bank_admins, _, _ = make_test_users(clerks_per_prison=1)
+        user_data = {
+            'username': 'new-bank-admin',
+            'first_name': 'New',
+            'last_name': 'Bank Admin',
+            'email': 'nba@mtp.gov.uk'
+        }
+        response = self.client.post(
+            self.get_url(),
+            format='json',
+            data=user_data,
+            HTTP_AUTHORIZATION=self.get_http_authorization_for_user(bank_admins[0])
+        )
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(len(User.objects.filter(username=user_data['username'])), 0)
+
+    def _check_create_user_succeeds(self, requester, user_data, client_id, groups):
+        self.client.post(
+            self.get_url(),
+            format='json',
+            data=user_data,
+            HTTP_AUTHORIZATION=self.get_http_authorization_for_user(requester)
+        )
+
+        new_user = User.objects.get(**user_data)
+        self.assertEqual(
+            new_user.applicationusermapping_set.all()[0].application.client_id,
+            client_id
+        )
+        self.assertEqual(
+            list(new_user.groups.all()),
+            groups
+        )
+        if hasattr(requester, 'prisonusermapping'):
+            self.assertEqual(
+                list(new_user.prisonusermapping.prisons.all()),
+                list(requester.prisonusermapping.prisons.all())
+            )
```

Comment:
again, not so much of an issue for tests yet, but the mappings have no specific order so comparing sets (of prison-user pairs) would work better than lists (of models)